upup/pkg/fi/cloudup/openstack/instance.go

```diff
@@ -91,6 +100,20 @@ func createInstance(c OpenstackCloud, opt servers.CreateOptsBuilder, portID stri
 			return false, fmt.Errorf("error creating server %v: %v", opt, err)
 		}
 		server = v
+
+		// TODO: 30 is such a magic number, configurable? or just const?
+		err = servers.WaitForStatus(c.ComputeClient(), server.ID, "ACTIVE", 30)
```

Comment:
when we create the server it should go from BUILD -> ACTIVE, if it goes anything else than ACTIVE we should fail this without waiting?

https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/util.go#L9 this will poll until the server is in ACTIVE and will ignore the ERROR state (when we could already exit the function).

Reply:
but ok, 30 secs is quite fast anyway.. in our case 30 secs is not maybe enough of waiting

Reply:
> when we create the server it should go from BUILD -> ACTIVE, if it goes anything else than ACTIVE we should fail this without waiting?
> 
> https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/util.go#L9 this will poll until the server is in ACTIVE and will ignore the ERROR state (when we could already exit the function).

yep, that would probably make more sense: waiting until the new server transitions away from `BUILD` and then see what state it's in:
* if `ACTIVE` everything's fine
* if other state (probably only `ERROR` possible in this case?), then fail

in some cases transitioning from `BUILD` to `ACTIVE` or `ERROR` will take some time though - we have some overloaded OpenStack clouds where our CI system provisions lots of servers and provisioning sometimes takes longer than 30secs - so "failing immediately" is relative. :)

I'll try to cook something up in that regard, and maybe also make the waiting for the state optional (feature flag?), so that the current way of provisioning will not be changed (someone may rely on that behavior?).